components/sync15/src/lib.rs

```diff
@@ -43,6 +39,9 @@ pub use crate::sync_multiple::{
     sync_multiple, sync_multiple_with_command_processor, MemoryCachedState, SyncRequestInfo,
 };
 pub use sync15_traits::client::DeviceType;
+pub use sync15_traits::SyncTraitsError;
```

Comment:
Curious why `sync15` needs to expose the error?

I think I'm missing something, aren't all the `sync15_traits::SyncTraitsError`s converted to `sync15::Error` before they are returned from any of the APIs?

If a consumer is depending on `sync15_traits` directly, it makes sense to me that it's up to the consumer to import the error through `sync15_traits` as opposed to importing it through `sync15`

Reply:
I guess I'm confused on "Why can't a consumer 'just' consume the types from sync15_traits through sync15_traits" (and subsequently, 'just' handle the error resulting from those types, as opposed to getting the types and errors from `sync15`, then dealing with two errors that `sync15` returns

Reply:
That's a good question :) I think it's because that `Into()` implementation isn't known about in tabs. I was indeed hoping it would be that simple though, so feel free to have a poke and see if you can improve that!

Reply:
(and I'll have another poke at that too, but later when I'm closer to landing it)

Reply:
> I think I'm missing something, aren't all the sync15_traits::SyncTraitsErrors converted to sync15::Error before they are returned from any of the APIs?

For a little more detail here - even though the functions exposes by sync15-traits are re-exported by sync15, their signature is still returning the `SyncTraitsError` - so I'm not surprised they aren't magically converted by the re-exported function. So we'd either need thin-wrappers in sync15 that just converted the error, *or* arrange for the conversion in sync15 to magically end up being used when (say) tabs sees the `SyncTraitsError` - it's that second option I was hoping we'd kinda get for free and I suspect the reason why we don't is that the converter in sync15 isn't seen or isn't able to be magically used.

Reply:
One source of magic would be a blanket impl, but I don't think it's possible.  I tried it 2 ways and both were rejected:
  - `sync15` could have  `impl<T> From<SyncTraitsError> for T where T: From<Error>`, but this violates the orphan rule.  Both `T` and `SyncTraitsError` are from other crates.
  - `sync15-traits` could have `impl<T, U> From<SyncTraitsError> for T where T: From<U>, U: From<SyncTraitsError>`,  but then you get the error `U` is an "unconstrained type parameter".  I think the issue here is essentially the same, it's possible for multiple types that satisfy the conditions for `U`.

I think a macro would work but that seems like overkill.